### PR TITLE
User login and protect route (need token to authenticate) 

### PR DESCRIPTION
**Pull request by tnlong1997**
* User login: get email and password from request, compare it with database. If correct, return {success, token}; else return {fail, err}
- Wrong password
screen shot 2018-04-10 at 12 19 27 am 2

- Correct email and password
screen shot 2018-04-10 at 12 19 19 am 2

* Add tokenAuthentication middleware for protect route
* Protect route: every route in protect routes need jwt in header of request to authenticate. If no jwt, return with err; else if not verify, return with err; else next to protect route api call

- No jwt: 

screen shot 2018-04-10 at 12 19 50 am 2

- Wrong jwt:

screen shot 2018-04-10 at 12 31 04 am 2

- Correct jwt: 

screen shot 2018-04-10 at 12 19 46 am 2

### CODE REVIEW

File: pmserver/config/secret.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = 'super.super.secret.shhh';
```
Comment (hoanghonn) [inline pmserver/config/secret.js]: đm tên secret như lol =))))

File: pmserver/routes/protect.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+var express = require('express');
+var router = express.Router();
+
+var tokenAuthentication = require('../middlewares/tokenAuthentication');
+
+// Add tokenAuthentication as middleware
+router.use(tokenAuthentication);
+
+router.get('/', function(req, res, next) {
```
Comment (hoanghonn) [inline pmserver/routes/protect.js]: m có thể add middleware vào trong method calll luôn đó ( get('/', middleware , function())

**Comment by hoanghonn** [comment]
có mấy cái t comment là góp ý thôi chứ ko có phải lỗi gì :) còn đâu thì ok rồi good work 👍

File: pmserver/config/secret.js (hunk 1 above)
Comment (tnlong1997) [reply pmserver/config/secret.js]: để t đổi thành longdeptrai :)

**Pull request merged**